### ISSUE TITLE
[BUG]Unpredictable behavior of the main window.

### ISSUE THREAD

**Issue by AirMarty** [section83/MacYTDL#34]
Configuration : 
macOS Mojave 
Version 10.14.6 (18G103)
iMac (27 pouces, fin 2013)
Proc 3,5 GHz Intel Core i7
Memory 16 Go
Dual Screen (ext screen via Thunderbolt to HDMI)

It seems the main window position change regularly during use, sometimes upper left corner, then in the middle of the screen, after downloading it will appear upper right corner, it can even appear with menu bar outside of the main display screen forcing abnormal manipulation.

**Comment by section83** [comment]
Apologies. A while back, I had to borrow a screen to try to fix the behaviour on dual screen setups. I didn’t do the best of work on that. Anyway, on first use of MacYTDL, the Main dialog always opens on the Main display.  Ordinarily, when the Main dialog is closed to Quit or to open Settings or Utilities, its position is saved (in MacYTDL.plist in preferences). Subsequently, the Settings and Utilities dialogs open at that position as does Main when it is reopened. The Main dialog’s position is always saved when it is closed. If it is not moved, it should always open in the same location.

Things can go haywire on dual setups e.g. if the main display is changed or repositioned while MacYTDL is open.

I’ll have another go at the logic. It might take a while if I have to borrow another screen.

**Comment by zxzzz8** [comment]
have the same issue all the time, but sense it seem to me somewhat related to https://github.com/section83/MacYTDL/issues/31 where you said you plan to change the tool you used to make these simple windows w....

so thats why i thought it might be unnecessary of me to mention until you fix that, but ye i recently had one time the app opened w a problematic location that i couldn't move it from until i restarted it....

![Screen Shot 2021-05-07 at 07;19;41](https://user-images.githubusercontent.com/20977838/117558737-77fc5b00-b088-11eb-9993-2394b04f00d0.png)

![Screen Shot 2021-05-07 at 07;17;56](https://user-images.githubusercontent.com/20977838/117558738-79c61e80-b088-11eb-91a7-770d682db674.png)

**Comment by zxzzz8** [comment]
oh but i have only one screen, an external screen for a mac mini.... 
also to me the main window and settings window always spawn somewhere in the left side of the screen i think? while the downloading message somwhere at the r side.... but ye still random where, everytime i open the app again....

**Comment by zxzzz8** [comment]
ok i just had it reset to that corner position like in the screenshots again, but not at app start but after putting a download.... so now i can't change the location until the download is over r? or is there some way? XD

**Comment by section83** [comment]
Blimey, looks like I can't "leave it till the next version" !  I'll have to borrow a screen and do some testing.

The loss of the title bar of a MacYTDL dialog signals that there has been a change in the screen layout since MacYTDL was opened.  Screen layout is usually changed in macOS `System Preferences => Displays`.  It includes changing which screen is the "main" display – i.e. the display showing the menu bar.  <= That is what I believe. But, the behaviour might be different in different versions of macOS and there are probably other ways to change the layout of screens.  This is at the limit of my knowledge at present.

MacYTDL saves its dialog position as relative to the screen layout at the moment it is closed (done by code in the _DialogToolkitMacYTDL_ script library).  So, if the layout is changed, the position at which MacYTDL dialogs display will also change.  Also, if you move the Main Dialog and press "Help", "Utilities", "Quit", "Settings" or "Continue", the new position will be saved.

No, you do not need to wait for downloads to finish.  Every download is a separate process, separate from each other and from MacYTDL's Main, Settings and Utilities dialogs.

Yes, the "Monitor" dialogs are different.  I decided to always position them in the top right hand corner of the main display.  That was because I wanted successive Monitor dialogs to be positioned without obscuring earlier Monitor dialogs (MacYTDL can start any number of separate downloads in succession – they run in parallel).  To make positioning easier, I fixed the position of the first Monitor dialog in the top right hand corner and further dialogs are shown in a column down the screen then across one dialog width to the left and down another column and so on for all the concurrent downloads underway.

Anyway, sorry you've had all that trouble.  I'll try to improve things in version 1.18.
